### ISSUE TITLE
po2php syntax issue

### ISSUE THREAD

**Issue by beachmat** [translate/translate#3283]
I'm trying to use php2po and po2php to translate some php string arrays like this:

```
<?php

return array(

	/*
	|--------------------------------------------------------------------------
	| String1
	|--------------------------------------------------------------------------
	|
	*/

  'string1'    => 'This is string1',
);
```

php2po generates a po file with the strings ok, and po2php seems to work, in that there are no error messages, even with errorlevel set to traceback, but it just gives me back the original untranslated file. Should this syntax be supported? If I change 
return array 
to 
$strings = array(..., 
it works ok. I'm using version 1.12 - would it work in version 1.13?
thanks

**Comment by unho** [comment]
Easier way to figure out is to try 1.13 :)

Have you tested converting the file without that comment within the array? Can you please provide a detailed list of the steps you follow to convert?

**Comment by beachmat** [comment]
Removing the comment makes no difference. I'd be willing to spend the time trying to install 1.13 if I knew it would fix the issue (I'm currently using 1.12 via Homebrew on Mac). 
What I did is php2po -P on original php to produce pot files, open them in POEdit, create new translation, save po file, then po2php -t to generate translated php. As I say, it works with $array = array(..., but not with return array(.

**Comment by beachmat** [comment]
Now tried TTK 1.13 and no difference from 1.12

**Comment by unho** [comment]
Confirmed the issue. It is a problem with `po2php` when trying to convert the unit back to PHP since the key we set in the PO unit is something it is unable to handle:

```po
#: return+array-%3E%27string1%27
msgid "This is string1"
msgstr ""
```

This is not an issue that can be quickly solved. For now the workaround that you confirmed it works is the only way to avoid this.

**Comment by perseusl** [comment]
Is this fixed already?? I'm also having the same problem

**Comment by unho** [comment]
No, it is not yet fixed. We have no immediate plans to fix, but we will try to get it fixed for next release (no date yet).

You say you need this to be fixed, but perhaps you can workaround for now using the trick described in the first message.

**Comment by unho** [comment]
@perseusl If you are a developer you might try replacing:

```php
return array(
```

with:

```php
$strings = array(
    ...
);

return $strings;
```

**Comment by bobhairgrove** [comment]
Using define('...') doesn't work, either. Here is my template PHP file:

>
?php
define('SBBL_ERROR_PAGE_TITLE','SBBL &mdash; Error');
define('SBBL_ERROR_PAGE_REDIRECT_MSG','In %d seconds you will be redirected to the %s page.');
define('SBBL_ERROR_PAGE_REDIRECT_MAIN','main');
define('SBBL_ERROR_PAGE_REDIRECT_LOGIN','login');
define('SBBL_ERROR_PAGE_HEADER','An error has occurred in the SBBL application:');
?>

And here is the German .PO file:

>
#. extracted from script_EN.php
msgid ""
msgstr ""
"Project-Id-Version: SBBL 1.0.0\n"
"Report-Msgid-Bugs-To: Robert Hairgrove <code@roberthairgrove.com>\n"
"POT-Creation-Date: 2016-07-06 17:38+0200\n"
"PO-Revision-Date: 2016-07-06 17:51+0100\n"
"Last-Translator: Robert Hairgrove <code@roberthairgrove.com>\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"X-Generator: Translate Toolkit 1.13.0\n"
"Language: de_DE\n"

>#: define('SBBL_ERROR_PAGE_TITLE'
msgid "SBBL &mdash; Error"
msgstr "SBBL &mdash; Fehlermeldung"

>#: define('SBBL_ERROR_PAGE_REDIRECT_MSG'
msgid "In %d seconds you will be redirected to the %s page."
msgstr "In %d Sekunden werden Sie zur %s umgeleitet."

>#: define('SBBL_ERROR_PAGE_REDIRECT_MAIN'
msgid "main"
msgstr "Hauptseite"

>#: define('SBBL_ERROR_PAGE_REDIRECT_LOGIN'
msgid "login"
msgstr "Login-Seite"

>#: define('SBBL_ERROR_PAGE_HEADER'
msgid "An error has occurred in the SBBL application:"
msgstr "Ein Fehler ist in der SBBL-Anwendung aufgetreten:"


Command line to generate new PHP file:
`po2php -t script_EN.php -i script_DE.po -o script_DE.php`

The resulting script_DE.php file is identical to the script_EN.php file.

**Comment by bobhairgrove** [comment]
PS ... I am using version 2.0.0b2
